test/common/schemaform/fields/ArrayField.unit.spec.jsx

```diff
@@ -239,4 +241,181 @@ describe('Schemaform <ArrayField>', () => {
+        showLastItemInViewMode: true,
+        viewField: f => f
+      }
+    };
+    const fullErrorSchema = {
+      0: {},
+    };
+    const onChange = sinon.spy();
+    const onBlur = sinon.spy();
+    const tree = SkinDeep.shallowRender(
+      <ArrayField
+        schema={schema}
+        errorSchema={fullErrorSchema}
+        uiSchema={uiSchema}
+        idSchema={idSchema}
+        registry={registry}
+        onChange={onChange}
+        onBlur={onBlur}
+        formData={formData}
+        formContext={formContext}
+        requiredSchema={requiredSchema}/>
+    );
+    expect(tree.everySubTree('SchemaField').length).to.equal(0);
+  });
+  it('should render edit mode when specified and has no data', () => {
+    const idSchema = {};
+    const schema = {
+      type: 'array',
+      items: [],
+      additionalItems: {
+        type: 'object',
+        properties: {
+          field: {
+            type: 'string'
+          }
+        }
+      }
+    };
+    const formData = [
+    ];
+    const uiSchema = {
+      'ui:title': 'List of things',
+      'ui:options': {
+        showLastItemInViewMode: true,
+        viewField: f => f
+      }
+    };
+    const fullErrorSchema = {
+    };
+    const onChange = sinon.spy();
+    const onBlur = sinon.spy();
+    const tree = SkinDeep.shallowRender(
+      <ArrayField
+        schema={schema}
+        errorSchema={fullErrorSchema}
+        uiSchema={uiSchema}
+        idSchema={idSchema}
+        registry={registry}
+        onChange={onChange}
+        onBlur={onBlur}
+        formData={formData}
+        formContext={formContext}
+        requiredSchema={requiredSchema}/>
+    );
+    expect(tree.everySubTree('SchemaField').length).to.equal(1);
+  });
+  it('should render edit mode when specified and has invalid data', () => {
+    const idSchema = {};
+    const schema = {
+      type: 'array',
+      items: [],
+      additionalItems: {
+        type: 'object',
+        properties: {
+          field: {
+            type: 'string'
+          }
+        }
+      }
+    };
+    const formData = [
+      {},
+    ];
+    const uiSchema = {
+      'ui:title': 'List of things',
+      'ui:options': {
+        showLastItemInViewMode: true,
+        viewField: f => f
+      }
+    };
+    const fullErrorSchema = {
+      0: { thing: { __errors: ['some error'] } },
+    };
+    const onChange = sinon.spy();
+    const onBlur = sinon.spy();
+    const tree = SkinDeep.shallowRender(
+      <ArrayField
+        schema={schema}
+        errorSchema={fullErrorSchema}
+        uiSchema={uiSchema}
+        idSchema={idSchema}
+        registry={registry}
+        onChange={onChange}
+        onBlur={onBlur}
+        formData={formData}
+        formContext={formContext}
+        requiredSchema={requiredSchema}/>
+    );
+    expect(tree.everySubTree('SchemaField').length).to.equal(1);
+  });
+  it('should render edit mode until submit if initial data incomplete or invalid', () => {
```

Comment:
@cvalarida this test has been set up for the bug you found.

Reply:
What do you think of the test description text `should not exit edit mode prematurely when validation errors are resolved`? It took me a couple tries to figure out what this test should be doing.

Reply:
How about `should not render review mode until valid data submitted` to clarify and try to keep the message short.